lib/queue.rb

```diff
@@ -1,31 +1,75 @@
 class Queue
 
+  MAX_SIZE = 20
+
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new
+    @front = @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if (@front == 0 && @back == MAX_SIZE - 1) || (@back == (@front -1) % MAX_SIZE - 1)
+      raise Exception.new("Queue is full")
+    elsif @store.empty? # queue is empty
+      @front = 0
+      @back = 0
+    elsif @back == (MAX_SIZE - 1) && @front != 0 # needs to wrap around
+      @back = 0
+    else # otherwise, ok to add
+      @back += 1
+    end
```

Comment:
I think you need to do `@back += 1` after the `@store[@back] = element` because otherwise you could add 1 element and front = back = 0 and then `self.empty?` would still be true.